packages/flutter_tools/lib/src/intellij/intellij_validator.dart

```diff
@@ -423,6 +434,43 @@ class IntelliJValidatorOnMac extends IntelliJValidator {
           }
         }
       }
+
+      final ProcessResult ceQueryResult = processManager.runSync(<String>[
+        'mdfind',
+        'kMDItemCFBundleIdentifier="com.jetbrains.intellij.ce"',
+      ]);
+
+      for (final String installPath in LineSplitter.split(ceQueryResult.stdout as String)) {
+        if (!validators.whereType<IntelliJValidatorOnMac>().any((IntelliJValidatorOnMac e) => e.installPath == installPath)) {
+          validators.add(IntelliJValidatorOnMac(
+            _communityEditionTitle,
+            _communityEditionId,
+            installPath,
+            fileSystem: fileSystem,
+            userMessages: userMessages,
+            plistParser: plistParser,
+            homeDirPath: homeDirPath,
+          ));
+        }
+      }
+
+      final ProcessResult ultimateQueryResult = processManager.runSync(<String>[
+        'mdfind',
+        'kMDItemCFBundleIdentifier="com.jetbrains.intellij*"',
```

Comment:
mdfind handles glob parsing?

Reply:
It actually has an entire query expression format.
https://developer.apple.com/library/archive/documentation/Carbon/Conceptual/SpotlightQuery/Concepts/QueryFormat.html#//apple_ref/doc/uid/TP40001849-CJBEJBHH